### PR TITLE
should be first allow_origins_by_regex

### PR DESCRIPTION
**Pull request by Qiuqiu0505**
### Description
allow_origins： Origins to allow CORS
allow_origins_by_regex： Regex to match with origin for enabling CORS.

As a user,  When I set the value of allow_origins_by_regex，the orgins to allow CORS must be in the range and other origin will  be failed. But it's not ready. 
In code，the default value of allow orgins  is  ‘*’ and all origins will be passed and allow_origins_by_regex has not effect.

In my opinion，allow_origins_by_regex should be considered first.  If the allow_origins_by_regex has no value. it can judge using allow_origins. 
Therefore, I submit the changes. Thanks.

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by tokers** [comment]
@Qiuqiu0505 Could you add some test cases to cover the changes?

**Comment by Qiuqiu0505** [comment]
> @Qiuqiu0505 Could you add some test cases to cover the changes?
 @tokers  Thanks for reminding and I have added some tests.

File: t/plugin/cors.t [hunk 1]
```diff
@@ -927,3 +927,83 @@ Access-Control-Allow-Headers: headr1,headr2
...
+--- response_body
+hello world
+--- response_headers
+Access-Control-Allow-Origin: http://sub.test.com
+Vary: Via, Origin
+Access-Control-Allow-Methods: GET,POST
+Access-Control-Allow-Headers: headr1,headr2
+Access-Control-Expose-Headers: ex-headr1,ex-headr2
+Access-Control-Max-Age: 50
+Access-Control-Allow-Credentials: true
```
Comment (tokers) [inline t/plugin/cors.t:1009]: Please take care the EOL.

Comment (Qiuqiu0505) [reply t/plugin/cors.t]: > Please take care the EOL.

Thanks, Is there anything else I need to add?

**Pull request closed without merging**